### ISSUE TITLE
Incoming call screen is shown underneath the fingerprint scanner

### ISSUE THREAD

**Issue by mijorus** [FossifyOrg/Phone#165]
### Checklist

- [X] I can reproduce the bug with the latest version given [here](https://github.com/FossifyOrg/Phone/releases/latest).
- [X] I made sure that there are **no existing issues** - [open](https://github.com/FossifyOrg/Phone/issues) or [closed](https://github.com/FossifyOrg/Phone/issues?q=is%3Aissue+is%3Aclosed) - to which I could contribute my information.
- [X] I made sure that there are **no existing discussions** - [open](https://github.com/FossifyOrg/Phone/discussions) or [closed](https://github.com/FossifyOrg/Phone/discussions?discussions_q=is%3Aclosed) - to which I could contribute my information.
- [X] I have read the FAQs inside the app (Menu -> About -> FAQs) and my problem isn't listed.
- [X] **I have taken the time to fill in all the required details. I understand that the bug report will be dismissed otherwise.**
- [X] This issue contains only one bug.
- [X] I have read and understood the [contribution guidelines](https://github.com/FossifyOrg/Phone/blob/master/CONTRIBUTING.md).

### Affected app version

1.1.1

### Affected Android/Custom ROM version

Android 14

### Affected device model

nothing phone 1

### How did you install the app?

Google Play Store

### Steps to reproduce the bug

Received a phone call with the screen locked

### Expected behavior

The UI should be interactive 

### Actual behavior

The incoming call UI is shown underneath the FP scanner

### Additional information

This is a long standing bug for this dialer app, I'm opening this hoping that will finally be fixed

**Comment by Guarmanda** [comment]
I have the same issue with lineageos. I have to unlock my screen before answearing

**Comment by da-anda** [comment]
same issue on a Moto Edge Neo 30 running their "stock" Android. On every incoming call one first has to unlock the screen using the default pin/pattern

**Comment by bugsbugsbux** [comment]
i have the same problem as #296 meaning the screen stays turned off (but it rings) and when i press the power button to wake the phone the call is terminated!

moto g 05, android 15

**Comment by eminfedar** [comment]
Same issue, pretty annoying, makes me lose couple of seconds while trying to answering the phone.

**Comment by naveensingh** [comment]
People on Xiaomi phones may try the following:

- Disabling battery restrictions for Fossify Phone in system settings (see [this comment](https://github.com/FossifyOrg/Phone/issues/108#issuecomment-1936318070))
- Enabling **Show on Lock screen** in **Other permissions**
- Solutions available on https://dontkillmyapp.com/xiaomi

<img alt="ignoreImageMinify" src="https://github.com/user-attachments/assets/d0489581-01df-4748-ae4b-15ac18ba5c3c" width=177 />

**Comment by GazaShaggy** [comment]
I also have this issue using Pixel 9 Pro and fdroid version

**Comment by da-anda** [comment]
I tried to find the cause for the issue and used AI to find possible root causes. After checking every suggestion by the AI, the last crucial requirement it gave for the fullscreen lock screen overlay to work was that the app HAS TO request the `USE_FULL_SCREEN_INTENT` intent for anything Android 12 and above. The current manifest does not seem to contain this.

Since this was suggested by AI, there is no guarantee it is true. But maybe a dev (@naveensingh ?) can verify or provide a test build?

Suggested code change by the AI:
```
<!-- 
CRITICAL: This permission is mandatory for Full-Screen Intents to work 
on devices running Android 12 (API 31) and higher.
Without this, the system ignores the setFullScreenIntent() call.
-->
<uses-permission android:name="android.permission.USE_FULL_SCREEN_INTENT" />

<!-- Example placement in the manifest: -->
<manifest xmlns:android="http://schemas.android.com/apk/res/android"
    package="org.fossify.phone">

    <!-- Add the USE_FULL_SCREEN_INTENT permission here -->
    <uses-permission android:name="android.permission.USE_FULL_SCREEN_INTENT" />

    <!-- ... existing permissions (e.g., READ_CONTACTS, CALL_PHONE) ... -->
    
    <application
        <!-- ... -->
    </application>
</manifest>

```

**Comment by naveensingh** [comment]
Yeah, that permission is needed. It could also be a faulty check in the code that decides when to show the activity (it currently does it  manually instead of relying on the system to pick between activity vs notification).

I remember running into some complication when I was fixing it a while back. I'll send a test build as soon as I'm home.

**Comment by naveensingh** [comment]
A test build with a potential fix is available [here](https://github.com/FossifyOrg/Phone/actions/runs/18729649588). Please let me know if it works.

**Comment by da-anda** [comment]
@naveensingh thanks much. Installed it on the affected phone and from a very brief test it seems to be working. I'll use it for a couple days and report back if the issue popps up again.

**Comment by GazaShaggy** [comment]
@naveensingh Thanks installed on mine as well and appears to function.

**Comment by eminfedar** [comment]
According to my tests, it is working too. Nothing Phone (1). Thanks @naveensingh . We are waiting an official release for this.

**Comment by WebworkrNet** [comment]
> A test build with a potential fix is available [here](https://github.com/FossifyOrg/Phone/actions/runs/18729649588/artifacts/4345109083). Please let me know if it works.

Just for your information:
The link currently leads to a 404 error.

If the test build was only available temporarily, it doesn't need to be restored as far as I'm concerned. I just noticed it.

**Comment by da-anda** [comment]
> > A test build with a potential fix is available [here](https://github.com/FossifyOrg/Phone/actions/runs/18729649588/artifacts/4345109083). Please let me know if it works.
> 
> Just for your information: The link currently leads to a 404 error.

build artifacts are usually not kept for long on github

**Comment by tswistak** [comment]
> Just for your information:
The link currently leads to a 404 error.

It works on my end. Maybe you have a misconfigured adblocker or something like this? Or maybe you're downloading on the device where you aren't logged into GitHub? Downloading artifacts works only after being logged in.

**Comment by WebworkrNet** [comment]
> > > A test build with a potential fix is available [here](https://github.com/FossifyOrg/Phone/actions/runs/18729649588/artifacts/4345109083). Please let me know if it works.
> > 
> > 
> > Just for your information: The link currently leads to a 404 error.
> 
> build artifacts are usually not kept for long on github

Thanks for the information! 👍

A ZIP file is now available for download again.

This fact is certainly not known to everyone and was also new to me.

@naveensingh
I would find it helpful if a note to this effect were included when setting build links in the future. :-)

**Comment by da-anda** [comment]
> > build artifacts are usually not kept for long on github
> 
> Thanks for the information! 👍
> 
> A ZIP file is now available for download again.
> 
> This fact is certainly not known to everyone and was also new to me.
> 
> [@naveensingh](https://github.com/naveensingh) I would find it helpful if a note to this effect were included when setting build links in the future. :-)

AFAIK projects can specify for how long. But I could also be mixing things up with Gitlab. My brain is a bit foggy atm (too much stress)

**Comment by WebworkrNet** [comment]
> > Just for your information:
> > The link currently leads to a 404 error.
> 
> It works on my end. Maybe you have a misconfigured adblocker or something like this? Or maybe you're downloading on the device where you aren't logged into GitHub? Downloading artifacts works only after being logged in.

I wasn't logged in before. 
That must have been the cause. 👍
Thanks

**Comment by naveensingh** [comment]
Thanks for testing!

**Comment by naveensingh** [comment]
> We are waiting an official release for this.

It's up now: https://github.com/FossifyOrg/Phone/releases/tag/1.8.0

> I would find it helpful if a note to this effect were included when setting build links in the future. :-)

@WebworkrNet You got it.

**Comment by da-anda** [comment]
thank you so much Naveen 👍. Much appreciated

**Comment by zanyshh** [comment]
> > We are waiting an official release for this.
> 
> It's up now: https://github.com/FossifyOrg/Phone/releases/tag/1.8.0
> 
> > I would find it helpful if a note to this effect were included when setting build links in the future. :-)
> 
> @WebworkrNet You got it.


Thank got the api updated to 36 🥰🥰

**Comment by TheFighter137** [comment]
If anyone is still having this issue with the latest updates and disabling battery optimization didn't work, here's another solution that worked on a Xiaomi : go into system settings > App list > Fossify Phone > Additional permissions, and turn on the ability to show up on the lock screen

**Comment by piterq** [comment]
> If anyone is still having this issue with the latest updates and disabling battery optimization didn't work, here's another solution that worked on a Xiaomi : go into system settings > App list > Fossify Phone > Additional permissions, and turn on the ability to show up on the lock screen

I've done this already and today I wasn't able to accept incoming call - accept and reject icons were displayed but none of them could be swiped (UI was blocked). After a while of struggling with the interface, I saw a lock screen where I had to enter the PIN and then I had the ability to answer the call.
I'm using Fossify Phone version 1.10.0 on Xiaomi 13T.

**Comment by ThomasRadeke** [comment]
I experience this very frequently, too, on a Nothing Phone (1), Android 15 and Fossify Phone 1.10 installed from F-Droid/IzzyOnDroid.
Phone rings and turns the screen on as expected, but I have to use the in-screen fingerprint reader to unlock the phone first.

I'm not sure if this might be connected to another issue I'm having, but I'll provide that for context:
- I often use a BT headset or the speakerphone for calling and the phone is propped up on a stand, facing me.
- I have face unlock and fingerprint set up.
- While in a call and the screen turn-off timeout kicks in, the screen briefly turns off, only to be woken up immediately, showing the fingerprint lock screen.
- Because I also have face unlock set up, the phone immediately unlocks and returns to the in-call screen.
- Even pressing the power button will not turn off the screen permanently - I have to obscure the top of the phone (proximity sensor and face unlock camera) to actually get it to stay off.

Again, I'm not sure if this might point to a common cause and I apologize if this doesn't really belong here, but I'll gladly make a new bug report if nobody else has reported this already.

**Comment by naveensingh** [comment]
It's being tracked in https://github.com/FossifyOrg/Phone/issues/676.

**Comment by chrispouliot** [comment]
I have recently installed the Fossify Phone app from f-droid on my Samsung Galaxy S25 updated to latest software. It seems like when I get a phone call, I have to unlock my phone with my pin or fingerprint to be able to see the answer/decline ringing screen. It seems to be intermittent. Is this issue resolved? I see the issue #676 seems specific to Xiaomi and about calls being declined, which is not the issue I am seeing. I do not see permissions for "Show on Lockscreen" in my OneUI app permission settings.
The Fossify Phone app is set as my default "Phone" app and has the following permissions: [Call logs, Contacts, Microphone, Notification, Phone] - in the All permissions menu it does say "display notifications as full screen activities on lock screen" is enabled.
Cheers

Fossify Phone Version 1.11.1
OneUI Version 8.0
Android 16

**Comment by WebworkrNet** [comment]
> > We are waiting an official release for this.
> 
> It's up now: https://github.com/FossifyOrg/Phone/releases/tag/1.8.0
> 
> > I would find it helpful if a note to this effect were included when setting build links in the future. :-)
> 
> [@WebworkrNet](https://github.com/WebworkrNet) You got it.

@naveensingh 
Note on downloading build files:
Very well done! :-)
Thanks

**Comment by Zhenchik88** [comment]
The issue is still present on **moto G84 5G**.

Fossify Phone Version 1.11.1
Android 15
